File: app/src/main/java/org/doomer/qnotez/utils/Dialogs.java
package org.doomer.qnotez.utils;

import android.app.Dialog;
import android.content.Context;
import android.util.Log;
import android.view.View;

import com.afollestad.materialdialogs.MaterialDialog;
import com.afollestad.materialdialogs.MaterialDialog.ListCallback;
import android.content.DialogInterface.OnClickListener;

import org.doomer.qnotez.R;

public class Dialogs {
    public static MaterialDialog createListDialog(final Context context, int strIdTitle, int strIdArrayList,
                                          ListCallback callback) {

        MaterialDialog.Builder builder = new MaterialDialog.Builder(context);
        builder.title(strIdTitle);
        builder.items(strIdArrayList);
        builder.itemsCallback(callback);

        MaterialDialog dialog = builder.build();
        return dialog;
    }

    public static MaterialDialog createConfirmDialog(final Context context,
                                                     int strIdTitle, int strIdContent,
                                                     MaterialDialog.SingleButtonCallback positive,
                                                     MaterialDialog.SingleButtonCallback negative) {
        MaterialDialog.Builder builder = new MaterialDialog.Builder(context);

        builder.title(strIdTitle);
        builder.content(strIdContent);
        builder.positiveText(context.getString(R.string.msg_yes));
        builder.negativeText(context.getString(R.string.msg_no));
        builder.onPositive(positive);
        builder.onNegative(negative);

        MaterialDialog dialog = builder.build();
        return dialog;
    }

    public static MaterialDialog createCustomSingleDialog(final Context context, int strIdTitle,
                                                          int custom_view) {

        MaterialDialog.Builder builder = new MaterialDialog.Builder(context);
        builder.title(strIdTitle);
        builder.customView(custom_view, false);

        MaterialDialog dialog = builder.build();
        return dialog;
    }
}

File: app/src/main/kotlin/org/doomer/qnotez/utils/Dialogs.kt
package org.doomer.qnotez.utils

import android.content.Context

import com.afollestad.materialdialogs.MaterialDialog
import com.afollestad.materialdialogs.MaterialDialog.ListCallback

import org.doomer.qnotez.R

object Dialogs {
    fun createListDialog(context: Context, strIdTitle: Int, strIdArrayList: Int,
                         callback: ListCallback): MaterialDialog {

        val builder = MaterialDialog.Builder(context)
        builder.title(strIdTitle)
        builder.items(strIdArrayList)
        builder.itemsCallback(callback)

        return builder.build()
    }

    fun createConfirmDialog(context: Context,
                            strIdTitle: Int, strIdContent: Int,
                            positive: MaterialDialog.SingleButtonCallback?,
                            negative: MaterialDialog.SingleButtonCallback?): MaterialDialog {
        val builder = MaterialDialog.Builder(context)

        builder.title(strIdTitle)
        builder.content(strIdContent)
        builder.positiveText(context.getString(R.string.msg_yes))
        builder.negativeText(context.getString(R.string.msg_no))

        positive?.let {
            builder.onPositive(positive)
        }

        negative?.let {
            builder.onNegative(negative)
        }

        return builder.build()
    }

    fun createCustomSingleDialog(context: Context, strIdTitle: Int,
                                 custom_view: Int): MaterialDialog {

        val builder = MaterialDialog.Builder(context)
        builder.title(strIdTitle)
        builder.customView(custom_view, false)

        return builder.build()
    }
}
